### ISSUE TITLE
[Console] - Special characters encoding issues

### ISSUE THREAD

**Issue by isaac-gros** [symfony/symfony#37278]
**Symfony version**
Symfony Console 5.0 (only this bundle). 

**Description**
I'm only using Symfony Console to generate PHP code and I have to ask informations to the user. When I run my script, I must use special characters to display a question, and it is displayed nicely. But when I submit informations with special characters, all of them are replaced by blank spaces.

**Example**
``` bash
Please write the value for the field "Libellé" : # The character rendering is fine there
> My favourite libellé # User input
The following libellé has been defined : My favourite libell  # Only the answer has been "sanitized"
```

**Possible solutions**
I tried to add `setTrimmable(false)` to the Question object, but it didn't help. I also tried to create a `validate` function, to see if I could get the input in it raw form, but the input is still sanitized. I also tried to save the input in a variable and convert it to UTF-8 but it couldn't change something about it. I tried the PHP functions `iconv` and `mb_convert_encoding` and none of them could solve the issue.

Do you guys have an idea about the issue origin and if there is a way to fix it ?

**Comment by isaac-gros** [comment]
Any idea ? 🤔

**Comment by YaFou** [comment]
Some issues are related to this problem: #36324 and #35842. This problem is caused by the function [`fgets`](https://www.php.net/manual/en/function.fgets.php) which does not take into account accents and special characters. I've not tested but [this answer](https://github.com/symfony/symfony/issues/36324#issuecomment-618186121) seems good. Are you on Windows @isaac-gros?

**Comment by isaac-gros** [comment]
> Some issues are related to this problem: #36324 and #35842. This problem is caused by the function [`fgets`](https://www.php.net/manual/en/function.fgets.php) which does not take into account accents and special characters. I've not tested but [this answer](https://github.com/symfony/symfony/issues/36324#issuecomment-618186121) seems good. Are you on Windows @isaac-gros?

Yes I am on Windows. I added `sapi_windows_cp_set(437);` in the main file that runs the console and it fixed the problem. If it can help anyone, my main file now looks like this :

``` php
#!/usr/bin/env php
<?php

require __DIR__.'/vendor/autoload.php';

use Symfony\Component\Console\Application;

sapi_windows_cp_set(437);

$application = new Application();
$application->run();
```
Thanks for the help @YaFou !

**Comment by YaFou** [comment]
> > Some issues are related to this problem: #36324 and #35842. This problem is caused by the function [`fgets`](https://www.php.net/manual/en/function.fgets.php) which does not take into account accents and special characters. I've not tested but [this answer](https://github.com/symfony/symfony/issues/36324#issuecomment-618186121) seems good. Are you on Windows @isaac-gros?
> 
> Yes I am on Windows. I added `sapi_windows_cp_set(437);` in the main file that runs the console and it fixed the problem. If it can help anyone, my main file now looks like this :
> 
> ``` php
> #!/usr/bin/env php
> <?php
> 
> require __DIR__.'/vendor/autoload.php';
> 
> use Symfony\Component\Console\Application;
> 
> sapi_windows_cp_set(437);
> 
> $application = new Application();
> $application->run();
> ```
> Thanks for the help @YaFou !

I've opened a pull request to fix this problem: #37385.

However be careful about calling the function `sapi_windows_cp_set` because only Windows supports this function. Use the following condition instead:
```php
if(function_exists(sapi_windows_cp_set)) {
    sapi_windows_cp_set(437);
}
```
